provider/google.go

```diff
@@ -182,7 +182,10 @@ func (p *googleProvider) DeleteZone(name string) error {
 func (p *googleProvider) Records(zone string) (endpoints []*endpoint.Endpoint, _ error) {
 	f := func(resp *dns.ResourceRecordSetsListResponse) error {
 		for _, r := range resp.Rrsets {
-			if r.Type != "A" {
+			switch r.Type {
+			case "A", "CNAME":
```

Comment:
It basically just filters out any records that aren't A or CNAME. As discussed with @ideahitme this can be removed once ownership is around. In the meantime this prevents some edge cases and other complexity but could lead to conflicts with existing records that have a different type.